e2e/api_test.go

```diff
@@ -24,21 +24,20 @@ import (
 func TestE2E_StatusEndpoints(t *testing.T) {
 	t.Parallel()
 
-	srv, err := newTestConsulServer(t)
-	require.NoError(t, err, "failed to start consul server")
+	srv := newTestConsulServer(t)
 	defer srv.Stop()
 
 	tempDir := fmt.Sprintf("%s%s", tempDirPrefix, "status_endpoints")
 	delete := testutils.MakeTempDir(t, tempDir)
 	// no defer to delete directory: only delete at end of test if no errors
 
-	configPath := filepath.Join(tempDir, configFile)
-
 	port, err := api.FreePort()
 	require.NoError(t, err)
 
-	err = makeConfig(configPath, fakeHandlerConfig(srv.HTTPAddr, tempDir, port))
-	require.NoError(t, err)
+	configPath := filepath.Join(tempDir, configFile)
+	config := fakeHandlerConfig().appendPort(port).
+		appendConsulBlock(srv).appendTerraformBlock(tempDir)
```

Comment:
Don't see this pattern much in Go but it works here very nicely. :100:

Reply:
+1 this cleaned up the `config.go` nicely!